### DISCUSSION TITLE
VScode power tools

### DISCUSSION THREAD

**Discussion by Rodsevich** [serverpod/serverpod#2167] [Ideas]
I want to facilitate the usage of Serverpod in the repo for newcomers, the features I need are:
 - hotreload
 - compound debug session of server and flutter dirs
 - code generation on background
**all with the press of the play button of VSCode**

I was wondering if the VSCode extension should include this or just the addition of `launch.json` and `tasks.json` on the root dir

**Comment by Katekko** [comment]
Hotreload: Easily manage server restarts by initiating debug mode and simply clicking the "Hotrestart" button. It's far more convenient than manually typing commands into the terminal each time. You can add "args": ["--apply-migrations"]  to your launch.json to facilitate even more.

Automatic Code Generation: Navigate to the server directory and execute serverpod generate --watch. This command ensures continuous monitoring of your server's changes, automatically generating the required code whenever modifications are made.

I typically prefer to keep my projects separated, with one window dedicated to the server and another to Flutter. This setup makes it easier for me to start, debug, and code efficiently.
